ttf2mesh.c

```diff
@@ -1151,18 +1157,23 @@ static int ttf_extract_tables(const uint8_t *data, int size, pps_t *s)
     rec = (ttf_tab_rec_t *)(s->hdr + 1);
 
     #define check_tag(str) (*(uint32_t *)rec->tableTag != *(uint32_t *)str)
-    #define match(type, name, str) \
+    #define match(type, name, str)                      \
     if (*(uint32_t *)rec->tableTag == *(uint32_t *)str) \
-    { \
-        s->s##name = rec->length; \
-        s->p##name = (type)(data + rec->offset); \
+    {                                                   \
+        s->s##name = rec->length;                       \
+        s->p##name = (type)(data + rec->offset);        \
+        if (rec->offset > INT_MAX - rec->length)        \
+            return TTF_ERR_FMT;                         \
+        if (rec->offset + rec->length > (unsigned)size) \
+            return TTF_ERR_FMT;                         \
     }
 
     while (ntab--)
     {
         conv32(rec->checkSum);
         conv32(rec->offset);
         conv32(rec->length);
+        if (rec->offset > INT_MAX - rec->length) return TTF_ERR_FMT;
```

Comment:
Integer overflow check

Reply:
You're right. The following check is probably the best way to check for overflow:
`if (offset + 4 > s-> scmap || offset + 4 < 0) return TTF_ERR_FMT;`